### ISSUE TITLE
Revise docs to start with bare composition YAMLs instead of packages

### ISSUE THREAD

**Issue by muvaf** [crossplane/crossplane#2322]
### What problem are you facing?

When users start reading our getting started docs, they start with installing a `Configuration` that contains XRDs amd `Composition`s. Then they move on to creation of the infra using that abstraction because we wanted to provide the fastest way to get users use composition.

However, we're getting feedback that this flow might be causing people assume that the packages are the main way to interact with Crossplane compositions, i.e. you always have to (or that's the recommended way) build & push a `Configuration` and then pull it into your cluster for each dev & test cycle they do. But this is not the case, you can just have your `Composition` and `CompositeResourceDefinition` YAMLs and edit them until you get to a point where you're ready to package them.

Another feedback is that some people find this flow too fast; we're getting them to see it all work with very little explanation of concepts and then drop the whole concepts docs after that.

### How could Crossplane help solve your problem?

The order we have is like the following today:
1. Install & Configure
  * Crossplane & CLI installation
  * Install Configuration
  * Configure Credentials
1. Provision Infrastructure
  * Create claim
  * Use the infra in a `Pod`
1. Create a Configuration
  * Create a Configuration Directory
  * Create CompositeResourceDefinition
  * Create Compositions
  * Build and Push The Configuration

I think instead of making the packages the main focus of the docs, we can focus more on the composition and keep the packaging to the last step. Something like the following:

1. Install & Configure
  * Crossplane & CLI installation
  * **Install Provider**
  * Configure Credentials
1. **Configure a Composition**
  * **Create XRD**
  * **Create `Composition`**
  * **Create claim and use it in a `Pod`**
1. Create a Configuration
  * Create a Configuration Directory with the XRD and `Composition` created earlier
  * Build and Push The Configuration

**Comment by negz** [comment]
Are we at a place where we can actively gather and consolidate feedback from the community about our docs before another refactor? I like the idea of evolving the docs, but it feels like we do a big reordering/refactor once every 2-3 releases and I'm not convinced we're reacting to strong signals vs reacting to 1-2 data points each time.

I see the point in explaining composition sooner, but this seems like such a subjective thing to me. I suspect some folks will want a "just let me see it in action, then dig deeper" flow while others will want more concepts explained from the start.

**Comment by negz** [comment]
Cross-linking to the last iteration of changes: https://github.com/crossplane/crossplane/pull/2166

**Comment by stevendborrelli** [comment]
+1 on @negz comment. I feel like we should do some more research with users to see how they approach the documentation and what roadblocks they face.

**Comment by turkenh** [comment]
Gathering more feedback would be good, however, I would already agree with the problem statement. 

> However, we're getting feedback that this flow might be causing people assume that the packages are the main way to interact with Crossplane compositions, i.e. you always have to (or that's the recommended way) build & push a Configuration and then pull it into your cluster for each dev & test cycle they do. But this is not the case, you can just have your Composition and CompositeResourceDefinition YAMLs and edit them until you get to a point where you're ready to package them.

FWIW, the above statement sounds like trying to teach k8s concepts by instructing how to package & deploy resources with helm :)

**Comment by negz** [comment]
> FWIW, the above statement sounds like trying to teach k8s concepts by instructing how to package & deploy resources with helm :)

I think if we just stopped at deploying the resources then never explained them that would be a problem. However in many cases folks like to see the thing working _first_ to help them understand the goal and see the value, then proceed to see how it was done.

Again - this is all super subjective so if at all possible I'd like us to try understand the best flow for the most people.

**Comment by muvaf** [comment]
> it feels like we do a big reordering/refactor once every 2-3 releases

FWIW, the last [iteration](https://github.com/crossplane/crossplane/pull/2166) didn't make changes regarding the issues mentioned here. I had some [concerns](https://github.com/crossplane/crossplane/pull/2166#pullrequestreview-601218392) back then as well. When you look at different versions of the docs since we introduced compositions, we never had a `Quick Start to Composition` guide since [v0.12](https://crossplane.io/docs/v0.12/getting-started/publish-infrastructure.html) when XRDs were called `InfrastructureDefinition`.

> However in many cases folks like to see the thing working first to help them understand the goal and see the value, then proceed to see how it was done.

I can see this being the case and I'm not saying that explaining everything first is the best way. As you point out, that's pretty subjective. However, the way the docs are structured make it sound like they have to use `Configuration`s to use composition. For example, if we applied a bunch of YAMLs with a single `kubectl apply`, let them see how it all works out and only then explain those YAMLs, that's also fine. Even letting them know that they don't have to package the YAMLs in a `Configuration` before deploying to the cluster at some point in the existing docs would go a long way.
